classes/scanner.php

```diff
@@ -58,49 +56,60 @@ public function is_configured() {
      */
     public function scan_file($file, $filename) {
         $this->post_file($file, $filename);
+        // Handle naughty statuses.
         if ($this->status === \core\antivirus\scanner::SCAN_RESULT_ERROR) {
             $this->message_admins(get_string('errorscanfile', 'antivirus_remote'));
             $this->set_scanning_notice(get_string('errorscanfile', 'antivirus_remote'));
         } else if ($this->status === \core\antivirus\scanner::SCAN_RESULT_FOUND) {
             $this->message_admins($this->response->msg);
             $this->set_scanning_notice($this->response->msg);
         }
+        // Return the status, we know this is mapped to the correct status constant inside of post_file.
         return $this->status;
     }
 
     /**
      * Post the file as form data to the remote engine.
      *
      * @param string $file location of the file
-      * @return void
+     * @param string $filename the name of the provided file.
+     * @param \curl $curl the curl object to use. Used for testing.
+     * @return void
      */
-    protected function post_file($file, $filename) {
+    protected function post_file($file, $filename, $curl = null) {
```

Comment:
> The uses of type hints and return type declarations is required in PHP in all possible locations for all new code. There will be necessary exclusions, such as code extending existing non-compliant code and implementing things where it is not available. Progressive approach will be applied. 

Can you please add type-hints to the function/method parameters here?

References:
- https://docs.moodle.org/dev/Coding_style#Functions_and_Methods